### ISSUE TITLE
Feedback on WSL2

### ISSUE THREAD

**Issue by philipp-baumann** [b-rodrigues/rix#6]
Hey @b-rodrigues , I just tested it on my friend's gaming machine using WSL2. Most hassle acutually came from installing the required system libraries. I will get back to you with a list of apt commands to install the dependencies of rix.

**Comment by b-rodrigues** [comment]
Interesting, I guess it's httr and jsonlite that require libcurl-dev and libopenssl-dev

**Comment by philipp-baumann** [comment]
> Interesting, I guess it's httr and jsonlite that require libcurl-dev and libopenssl-dev

exactly. I was wondering if it makes more sense to install everything on win and then go to the mounted C or whatever drive though Ubuntu WSL2?

**Comment by b-rodrigues** [comment]
Honestly it might be easier to install `rix()` using Nix as well 😄 essentially what is described here: https://b-rodrigues.github.io/rix/articles/interactive-use.html#scenario-2-you-install-r-and-rstudio-using-nix

I would need to make these instructions more prominent

**Comment by philipp-baumann** [comment]
aah good idea!

**Comment by philipp-baumann** [comment]
#7

**Comment by b-rodrigues** [comment]
yeah I guess a note in the readme redirecting users to these instructions would be needed, but I'd like to find out what should be installed on WSL2 as well. My guess is that these libraries should be installed within the Ubuntu within Windows ?

**Comment by philipp-baumann** [comment]
> yeah I guess a note in the readme redirecting users to these instructions would be needed, but I'd like to find out what should be installed on WSL2 as well. My guess is that these libraries should be installed within the Ubuntu within Windows ?

yes, the link would be good. But then actually I am not sure if one can not just navigate to the C drive, which is available on WSL2, too?

**Comment by philipp-baumann** [comment]
> > yeah I guess a note in the readme redirecting users to these instructions would be needed, but I'd like to find out what should be installed on WSL2 as well. My guess is that these libraries should be installed within the Ubuntu within Windows ?
> 
> yes, the link would be good. But then actually I am not sure if one can not just navigate to the C drive, which is available on WSL2, too?

Maybe doing both would resolve? Meaning a short description of sys libs on WSL2 plus the way to just use rix on windows and then navigate to the mounted drive?

**Comment by b-rodrigues** [comment]
yes I guess so

**Comment by philipp-baumann** [comment]
@b-rodrigues hmm unfortunately the azure VMs do not support virtualization in bios and hence it is not possible to run WSL2 on them, as far as I can tell.

**Comment by b-rodrigues** [comment]
yes, I believe that would be enough, plus hinting at the option of bootstrapping everything using the provided default.nix file from the documention.

**Comment by philipp-baumann** [comment]
good idea.

**Comment by philipp-baumann** [comment]
#12

**Comment by philipp-baumann** [comment]
@b-rodrigues is there a default `.nix` file you ship with the pkg. Did not find using `ripgrep`

**Comment by b-rodrigues** [comment]
no, the structure of the file is defined here: https://github.com/b-rodrigues/rix/blob/5b0c962823c38476f2b2799f8483f5dc0e4f3698/dev/build_envs.Rmd#L398

**Comment by philipp-baumann** [comment]
> no, the structure of the file is defined here:
> 
> https://github.com/b-rodrigues/rix/blob/5b0c962823c38476f2b2799f8483f5dc0e4f3698/dev/build_envs.Rmd#L398

perfect thanks!

**Comment by philipp-baumann** [comment]
Would say we keep this open to have a place where Windows users can bring up stuff :-)

**Comment by philipp-baumann** [comment]
New plan:

- `nix_build()`, if on Windows native host, and if WSL is present, then preprend `wsl` to the build cmd.
